keps/sig-scheduling/20200612-simplified-topology-manager.md

```diff
@@ -0,0 +1,229 @@
+- This Proposal requires exposing NUMA topology information. This KEP doesn't
+describe how to expose all necessary information it just declare what kind of
+information is necessary.
+
+# Proposal
+Kube-scheduler builtin plugin will be added to the main tree. This plugin
+implements a simplified version of TopologyManager it’s different from original topology manager algorithm.
+Plugin checks the ability to run pod only in case of single-numa-node policy on the
+node, since it is the most strict policy, it implies that the launch on the node with
+other existing policies will be successful if the condition for single-numa-node policy passed for the worker node.
+Proposed plugin will use [CRD][1] to identify which topology policy is enabled on the node.
+To work, this plugin requires topology information of the available resource on the worker nodes.
+
+## Topology format
+
+Available resources with topology of the node should be stored in CRD. Format of the topology described
+[in this document][1].
+
+The daemon which runs outside of the kubelet will collect all necessary information on running pods, based on allocatable resources of the node and consumed resources by pods it will provide available resources in CRD, where one CRD instance represents one worker node. The name of the CRD instance is the name of the worker node.
+
+## Plugin implementation details
+
+Since topology of the node is stored in the CRD, kube-scheduler should be subscribed for updates of appropriate CRD type. Kube-scheduler will use informers which will be generated with the name NodeTopologyInformer. NodeTopologyInformer will run in TopologyMatch plugin.
+
+### Topology information in the TopologyMatch plugin
+
+Once NodeResourceTopology is received TopologyMatch plugin keeps it in its own state of type
+NodeTopologyMap. This state will be used every time when scheduler needs to make a decidion based on node topology.
+
+```go
+
+type NodeTopologyMap map[string]topologyv1alpha1.NodeResourceTopology
+
+type NodeResourceTopology struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    TopologyPolicy string        `json:"topologyPolicy"`
+    Nodes   []NUMANodeResource   `json:"nodes"`
+}
+
+// NUMANodeResource is the spec for a NodeResourceTopology resource
+type NUMANodeResource struct {
+    NUMAID int
+    Resources v1.ResourceList
+}
+```
+Where TopologyPolicy may have following values: none, best-effort, restricted, single-numa-node
+
+To use these labels both in kube-scheduler and in kubelet string constants of these labels should be moved from pkg/kubelet/cm/topologymanager/ and pkg/kubelet/apis/config/types.go to pkg/apis/core/types.go a one single place.
+
+NUMAID is an auxiliary field since scheduler version of TopologyManager doesn't make a real assignment.
+
+### Description of the Algorithm
+
+The algorithm which implements single-numa-node policy is following:
+
+```go
+if !guaranteedQoS(pod) {
```

Comment:
the topology manager aligns pods of all QoS classes.

see:
https://github.com/kubernetes/kubernetes/pull/83492

Reply:
Thanks for letting us know, weren't aware of the changes made by this PR.